### PR TITLE
Refactor import checking to read from bytecode

### PR DESCRIPTION
**Pull request by kjdart**
SHLOMO-155

### CODE REVIEW

File: shlomobot_pytest/common_tests.py [hunk 1]
```diff
@@ -18,9 +18,9 @@
     function_contains_regex,
     get_function_regex_matches,
 )
-
 # ================= CONSTANTS =================
 
+GLOBAL_DECLERATION_REGEX = re.compile(r"\n\s*global\s+[^\W]+(?:\s*,\s*[^\W]+\s*)*\n")
```
Comment (KeshetS) [inline shlomobot_pytest/common_tests.py:23]: Why this was changed?

File: shlomobot_pytest/common_tests.py [hunk 2]
```diff
@@ -159,6 +159,24 @@ def function_is_one_liner(py_filename: str, function_name: str) -> bool:
     return False
 
 
+def correct_imports_are_made(module_name: str, import_list: list[str]) -> bool:
+    """Checks if the all the required modules from import_list have been imported"""
```
Comment (KeshetS) [inline shlomobot_pytest/common_tests.py:163]: Well done. Just add a comment that it doesn't check if unnecessary imports were made.

**Review by KeshetS — CHANGES_REQUESTED** [review]
Good job!

File: shlomobot_pytest/common_tests.py (hunk 1 above)
Comment (kjdart) [reply shlomobot_pytest/common_tests.py]: I think I made an error when merging in the main branch. I've removed it in the latest commit

File: shlomobot_pytest/common_tests.py (hunk 2 above)
Comment (kjdart) [reply shlomobot_pytest/common_tests.py]: I've added the comment in the latest commit

**Review by PHAng-DART — COMMENTED** [review]
I am unsure if the correct_imports_are_made function will have the same problem as the function_contains_inputs function whereby the different python versions will generate different opcodes.

**Review by KeshetS — APPROVED** [review]

**Pull request merged**